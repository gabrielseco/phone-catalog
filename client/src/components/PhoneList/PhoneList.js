// @flow
import React from 'react';
import { PhoneListItem } from 'components';
import { type Phone } from './../../types/phone';
import styles from './PhoneList.scss';

type PhoneListProps = {
  phones: Phone[]
};

const PhoneList = (props: PhoneListProps) => {
  const phones = props.phones.map((phone, index) => {
    return <PhoneListItem phone={phone} key={index} />;
  });

  return (
    <div role="list" className={styles.container}>
      {phones}
    </div>
  );
};

export default PhoneList;
