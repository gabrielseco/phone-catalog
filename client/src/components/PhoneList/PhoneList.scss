@import '~styles/settings/global';
@import '~styles/tools/mixins/all';

.container {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  @include respond-to(medium) {
    flex-direction: row;
  }
}
