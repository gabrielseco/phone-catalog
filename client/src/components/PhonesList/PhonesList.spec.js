import React from 'react';
import { shallow } from 'enzyme';

import PhoneList from './PhonesList';

describe('PhoneList suite', () => {
  it('should render the ProductsList component', () => {
    const component = shallow(<PhoneList phones={[]} />);
    expect(component).toBeDefined();
  });
});
