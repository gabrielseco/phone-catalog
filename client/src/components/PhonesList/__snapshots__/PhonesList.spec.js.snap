// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneList suite should render the ProductsList component 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <img
      alt="Iphone X"
      className="phoneImg"
      src="https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg"
    />
    <h2
      className="title"
    >
      Iphone X
    </h2>
    <p
      className="price"
    >
      999$
    </p>
    <a
      className="btnAction"
      href="detail/1"
      onClick={[Function]}
    >
      See more
    </a>
  </div>
  <div
    className="container"
  >
    <img
      alt="Samsung Galaxy S9"
      className="phoneImg"
      src="https://www.satugadget.com.my/content/images/thumbs/0008898_samsung-galaxy-s9-plus-256gb-purple-original-malaysia-set_300.jpeg"
    />
    <h2
      className="title"
    >
      Samsung Galaxy S9
    </h2>
    <p
      className="price"
    >
      899$
    </p>
    <a
      className="btnAction"
      href="detail/2"
      onClick={[Function]}
    >
      See more
    </a>
  </div>
</div>
`;
